frontend/projects/abtesting/src/app/core/http-interceptors/http-error.interceptor.ts

```diff
@@ -4,25 +4,49 @@ import {
   HttpInterceptor,
   HttpHandler,
   HttpRequest} from '@angular/common/http';
-import { Observable, throwError } from 'rxjs';
+import { Observable, EMPTY } from 'rxjs';
 import { catchError } from 'rxjs/operators';
 import { AuthService } from '../auth/auth.service';
+import { environment } from '../../../environments/environment';
+import { NotificationService } from '../notifications/notification.service';
 
 @Injectable()
 export class HttpErrorInterceptor implements HttpInterceptor {
   constructor(
-    private authService: AuthService
+    private authService: AuthService,
+    private notificationsService: NotificationService
   ) { }
 
+  timeOut = 1500;
+  networkErrorCount: number = 0;
+
+  incrementErrorCount() {
+      return ++this.networkErrorCount;
+  }
+  
+  openSnackBar(error) {
+    let displayMessage = 'Network call failed for "' + error.url + '".';
+
+    if (!environment.production) {
+      displayMessage += ' See console for details.';
+    }
+    if (!((error as any).status === 401) && !environment.production) {
+      setTimeout(() => {
+        this.notificationsService.error(displayMessage); // this has 1500 as timeout
+      }, this.networkErrorCount * (this.timeOut+500)); // scheduling each snackbar one after the other with a gap of 500ms
```

Comment:
This seems incorrect. Correct me here but it should be
`(this.networkErrorCount * this.timeOut)+500`

Reply:
No, this is correct, if I do it as you mentioned the scheduling will be back to back without any gap of 500ms, only the first one would start after 500ms, rest would be exact next to first one.

Reply:
Yup got it!